harness/tests/integration_cases/test_raft.rs

```diff
@@ -4724,3 +4724,59 @@ fn test_request_snapshot_on_role_change() {
         nt.peers[&2].pending_request_snapshot
     );
 }
+
+/// Tests group commit.
+///
+/// 1. Logs should be replicated to at least different groups before committed;
+/// 2. If no groups has been configured, quorum should be used.
+#[test]
+fn test_group_commit() {
+    let l = default_logger();
+    let mut tests = vec![
+        // single
+        (vec![1], vec![0], 1),
+        (vec![1], vec![1], 1),
+        // odd
+        (vec![2, 2, 1], vec![1, 2, 1], 2),
+        (vec![2, 2, 1], vec![1, 1, 2], 1),
+        (vec![2, 2, 1], vec![1, 0, 1], 1),
+        (vec![2, 2, 1], vec![0, 0, 0], 2),
+        // even
+        (vec![4, 2, 1, 3], vec![0, 0, 0, 0], 2),
+        (vec![4, 2, 1, 3], vec![1, 0, 0, 0], 1),
+        (vec![4, 2, 1, 3], vec![0, 1, 0, 2], 2),
+        (vec![4, 2, 1, 3], vec![0, 2, 1, 0], 1),
+        (vec![4, 2, 1, 3], vec![1, 1, 1, 1], 1),
```

Comment:
So if 4 peers are in one group, to commit a log needs 4 ACKs? I think we can ignore group ID if there is only 1 group.